dspace-api/src/main/java/org/dspace/content/CollectionServiceImpl.java

```diff
@@ -831,4 +836,14 @@ public Collection findByIdOrLegacyId(Context context, String id) throws SQLExcep
     public Collection findByLegacyId(Context context, int id) throws SQLException {
         return collectionDAO.findByLegacyId(context, id, Collection.class);
     }
+
+    @Override
+    public int countTotal(Context context) throws SQLException {
+        return collectionDAO.countRows(context);
+    }
+
+    @Override
+    public List<Map> getCollectionsWithBitstreamSizesTotal(Context context) throws SQLException {
```

Comment:
Yes it would, is there a way how to get the generics from the Query object in the DAOImpl ? The map actually doesn't make much of a sense, I'd rather see `List<Map.Entry<Collection,Long>>`.
I thought having `SELECT NEW java.util.AbstractMap.SimpleEntry(col, sum(bit.sizeBytes)...)` in the query at https://github.com/DSpace/DSpace/pull/994/files#diff-5e6661ba14882609a90e197c2933a2f6R175 would accomplish that, but that didn't seem to be working.
But I could have at least copied the `List<Map>` to a new collection...